### ISSUE TITLE
Non-absolute path exception in EditDomainHandler.getAssists

### ISSUE THREAD

**Issue by fabitamas** [dart-lang/sdk#34152]
Analyzer Feedback from IntelliJ

# Version information

- `IDEA IU-182.3911.36`
- `2.0.0`
- `IU-182.3911.36, JRE 1.8.0_152-release-1248-b8x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.13.6 unknown, screens 1440x900, 1920x1080, 1920x1080  Retina`

# Exception

Please append the contents of:
file:///private/var/folders/t0/7zp2_4jn6z50rsrhzsh_yztj80zyr9/T/report.txt

```
Dart analysis server, SDK version 2.0.0, server version 1.20.3, FATAL error: Failed to handle request: {id: 408, method: edit.getAssists, params: {file: localhost:51928/packages/angular_components/app_component.template.dart, offset: 7533, length: 0}, clientRequestTime: 1534276220119}
Invalid argument(s): Only absolute paths are supported: localhost:51928/packages/angular_components/app_component.template.dart
#0      AnalysisDriver._throwIfNotAbsolutePath (package:analyzer/src/dart/analysis/driver.dart:1755:7)
#1      AnalysisDriver.getResult (package:analyzer/src/dart/analysis/driver.dart:771:5)
#2      AnalysisServer.getAnalysisResult (package:analysis_server/src/analysis_server.dart:622:10)
#3      EditDomainHandler.getAssists (package:analysis_server/src/edit/edit_domain.dart:175:42)
<asynchronous suspension>
#4      EditDomainHandler.handleRequest (package:analysis_server/src/edit/edit_domain.dart:342:9)
#5      AnalysisServer.handleRequest.<anonymous closure>.<anonymous closure> (package:analysis_server/src/analysis_server.dart:751:45)
#6      _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:210:15)
#7      AnalysisServer.handleRequest.<anonymous closure> (package:analysis_server/src/analysis_server.dart:747:50)
#8      _rootRun (dart:async/zone.dart:1124:13)
#9      _CustomZone.run (dart:async/zone.dart:1021:19)
#10     _runZoned (dart:async/zone.dart:1516:10)

**Comment by devoncarew** [comment]
Invalid argument(s): Only absolute paths are supported: localhost:51928/packages/angular_components/app_component.template.dart

```
#3      EditDomainHandler.getAssists
```

I think we can throw with `Only absolute paths are supported` on something like set analysis roots, but for the other calls, we may want to be more lenient about how we fail for non-absolute paths. In this case, we may want to return an error to the client for the actual call, instead of reporting an analysis server crash.

**Comment by bwilkerson** [comment]
I agree: all of the requests should return an error response if the inputs are invalid; they should _never_ crash.

But all of the file paths for all of the requests are expected to be absolute paths, and I don't think we want to change that.

**Comment by stereotype441** [comment]
Escalating to P1 because of recent duplicates

**Comment by stereotype441** [comment]
@scheglov any idea why this might be happening?

**Comment by stereotype441** [comment]
@bwilkerson both of those seem like problems to me.  I think we should address the latter problem first (since fixing the former problem would mask the bug).  Then we should work with @jwren to fix the former problem (note that IntelliJ is the client in this case).

**Comment by scheglov** [comment]
https://dart-review.googlesource.com/c/sdk/+/90840 for DAS.

**Comment by devoncarew** [comment]
Is there still an IDE issue here (sending in non-absolute paths)?

**Comment by bwilkerson** [comment]
Yes.

**Comment by bwilkerson** [comment]
But it might be better to open a separate issue for that and re-close this one.

**Comment by stereotype441** [comment]
Agreed!  @jwren would you mind filing an IntelliJ issue in the appropriate place and then closing this bug?

**Comment by jwren** [comment]
Done, https://github.com/dart-lang/sdk/issues/35741

**Comment by scheglov** [comment]
There are "lack of capacity" failures on bots.
I suspect, that this is because of this CL, so I'm reverting it.

**Comment by scheglov** [comment]
The CL (with minor Windows fixes) was re-landed as 6de1c478c3bbdc604e3d63c633a4686622429cf3.
